include/datatypes.h

```diff
@@ -134,7 +108,104 @@ typedef struct _session_id_tracker_t
     uint32_t          session_id;
 } session_id_tracker_t;
 
-#pragma pack(pop)
+//sgx-ssl framework
+
+typedef struct {
+    uint32_t keyspec;
+    uint32_t digest_mdoe;
+    uint32_t padding_mdoe;
```

Comment:
typo, mdoe->mode

Reply:
the modes are not common parameters, you can put them into a separate struct as "addition", such as defined them as:
typedef struct {
  uint32_t keyspec;
  uint32_t origin;
  uint32_t purpose;
  uint32_t apiversion;
  uint8_t  descrption[16];
  uint8_t createdate[8];
  struct addition addition;
}

you can try to find a better name.